### ISSUE TITLE
Compile error: BasicErrorManager in com.google.javascript.jscomp has been deprecated

### ISSUE THREAD

**Issue by carylewis** [google/closure-compiler#3535]
I am trying to run:

`mvn -DskipTests -pl externs/pom.xml,pom-main.xml,pom-main-shaded.xml
`
and mvn errors out with the following:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.5.1:compile (default-compile) on project closure-compiler: Compilation failure: Compilation failure: 
[ERROR] /home/centos/libphonenumber-tools/closure-compiler/src/com/google/javascript/jscomp/deps/JsFileFullParser.java:[27,35] [deprecation] BasicErrorManager in com.google.javascript.jscomp has been deprecated
```

I am running the latest mvn:

> mvn --version
> Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
> Maven home: /opt/maven
> Java version: 1.8.0_232, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.232.b09-0.el7_7.x86_64/jre
> Default locale: en_CA, platform encoding: UTF-8
> OS name: "linux", version: "3.10.0-1062.7.1.el7.x86_64", arch: "amd64", family: "unix"

I am trying to run this on a centos7 EC2 instance on AWS.

**Comment by carylewis** [comment]
Update:

I modified the pom.xml and removed the deprecation

`<compilerArgument>-Xlint:unchecked,deprecation,fallthrough,finally</compilerArgument>`

Now the build fails with:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project closure-compiler: Compilation failure -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project closure-compiler: Compilation failure
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.compiler.CompilationFailureException: Compilation failure
    at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute (AbstractCompilerMojo.java:1224)
    at org.apache.maven.plugin.compiler.CompilerMojo.execute (CompilerMojo.java:187)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:4
06) at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

I confirmed that I am running the jdk:

```
apt install openjdk-8-jdk
Reading package lists... Done
Building dependency tree       
Reading state information... Done
openjdk-8-jdk is already the newest version (8u232-b09-0ubuntu1~16.04.1).
0 upgraded, 0 newly installed, 0 to remove and 47 not upgraded.
```
Maven version:

```
mvn --version
Apache Maven 3.3.9
Maven home: /usr/share/maven
Java version: 1.8.0_232, vendor: Private Build
Java home: /usr/lib/jvm/java-8-openjdk-amd64/jre
Default locale: en_US, platform encoding: ANSI_X3.4-1968
OS name: "linux", version: "3.10.0-1062.7.1.el7.x86_64", arch: "amd64", family: "unix"

```

javac version:

```
javac -version
javac 1.8.0_232
```


The latest attempt to compile was using a Ubuntu 16.04 docker container:
```

lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.2 LTS
Release:	16.04
Codename:	xenial
```

**Comment by carylewis** [comment]
Is there any update? Am I the only one that can't compile this?

What environment is used to compile the project internally?

**Comment by niloc132** [comment]
I'm another external user, but not having an issue compiling on the current master. Your two posts differ in maven version, is that deliberate?

I do get warnings about BasicErrorManager being deprecated though:
```
...
[INFO] --- maven-compiler-plugin:3.5.1:compile (default-compile) @ closure-compiler ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 774 source files to /home/colin/workspace/closure-compiler/target/classes
[WARNING] /home/colin/workspace/closure-compiler/src/com/google/javascript/jscomp/ant/AntErrorManager.java:[19,35] [deprecation] BasicErrorManager in com.google.javascript.jscomp has been deprecated
[WARNING] /home/colin/workspace/closure-compiler/src/com/google/javascript/jscomp/deps/JsFileFullParser.java:[27,35] [deprecation] BasicErrorManager in com.google.javascript.jscomp has been deprecated
[WARNING] /home/colin/workspace/closure-compiler/src/com/google/javascript/refactoring/FixingErrorManager.java:[30,35] [deprecation] BasicErrorManager in com.google.javascript.jscomp has been deprecated
[WARNING] /home/colin/workspace/closure-compiler/src/com/google/javascript/jscomp/ant/AntErrorManager.java:[19,35] [deprecation] BasicErrorManager in com.google.javascript.jscomp has been deprecated
[WARNING] /home/colin/workspace/closure-compiler/src/com/google/javascript/jscomp/deps/JsFileFullParser.java:[27,35] [deprecation] BasicErrorManager in com.google.javascript.jscomp has been deprecated
[WARNING] /home/colin/workspace/closure-compiler/src/com/google/javascript/refactoring/FixingErrorManager.java:[30,35] [deprecation] BasicErrorManager in com.google.javascript.jscomp has been deprecated
[WARNING] /home/colin/workspace/closure-compiler/src/com/google/javascript/jscomp/ant/AntErrorManager.java:[19,35] [deprecation] BasicErrorManager in com.google.javascript.jscomp has been deprecated
[WARNING] /home/colin/workspace/closure-compiler/src/com/google/javascript/jscomp/deps/JsFileFullParser.java:[27,35] [deprecation] BasicErrorManager in com.google.javascript.jscomp has been deprecated
[WARNING] /home/colin/workspace/closure-compiler/src/com/google/javascript/refactoring/FixingErrorManager.java:[30,35] [deprecation] BasicErrorManager in com.google.javascript.jscomp has been deprecated
[WARNING] /home/colin/workspace/closure-compiler/src/com/google/javascript/jscomp/ant/AntErrorManager.java:[19,35] [deprecation] BasicErrorManager in com.google.javascript.jscomp has been deprecated
[WARNING] /home/colin/workspace/closure-compiler/src/com/google/javascript/jscomp/deps/JsFileFullParser.java:[27,35] [deprecation] BasicErrorManager in com.google.javascript.jscomp has been deprecated
[WARNING] /home/colin/workspace/closure-compiler/src/com/google/javascript/refactoring/FixingErrorManager.java:[30,35] [deprecation] BasicErrorManager in com.google.javascript.jscomp has been deprecated
...
```

One suggestion to make on your maven command line: be explicit about the goals you are running, and/or try a `clean` once, as you may have some stale content messing things up. To ensure that I was running the same as you, here's the command i'm running and some other details:
```
[colin@runes closure-compiler (master)]$ mvn --version
Apache Maven 3.6.3 (NON-CANONICAL_2019-11-27T20:26:29Z_root)
Maven home: /opt/maven
Java version: 1.8.0_242, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-8-openjdk/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.4.15-arch1-1", arch: "amd64", family: "unix"
[colin@runes closure-compiler (master)]$ git rev-parse HEAD
6f90cce5364789b51edd64fee2bebdb520b01d75
[colin@runes closure-compiler (master)]$ mvn -DskipTests -pl externs/pom.xml,pom-main.xml,pom-main-shaded.xml  clean install

...

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Closure Compiler Externs 0.1-SNAPSHOT:
[INFO] 
[INFO] Closure Compiler Externs ........................... SUCCESS [  0.781 s]
[INFO] Closure Compiler Main .............................. SUCCESS [  0.472 s]
[INFO] Closure Compiler ................................... SUCCESS [ 27.271 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  29.096 s
[INFO] Finished at: 2020-02-05T08:43:18-06:00
[INFO] ------------------------------------------------------------------------
```

**Comment by carylewis** [comment]
Thanks for the info, I tried to re-clone the repo, verified that I am on the same branch as you, and when I ran the  mvn -DskipTests -pl externs/pom.xml,pom-main.xml,pom-main-shaded.xml  clean install, I again got these errors:

```
[WARNING] /home/centos/closure-compiler/src/com/google/javascript/jscomp/deps/JsFileFullParser.java:[27,35] [deprecation] BasicErrorManager in com.google.javascript.jscomp has been deprecated
[WARNING] /home/centos/closure-compiler/src/com/google/javascript/jscomp/ant/AntErrorManager.java:[19,35] [deprecation] BasicErrorManager in com.google.javascript.jscomp has been deprecated
[INFO] 9 warnings 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Closure Compiler Externs 1.0-SNAPSHOT:
[INFO] 
[INFO] Closure Compiler Externs ........................... SUCCESS [  2.731 s]
[INFO] Closure Compiler Main .............................. SUCCESS [  3.378 s]
[INFO] Closure Compiler ................................... FAILURE [ 30.641 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  39.201 s
[INFO] Finished at: 2020-02-06T13:45:06Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.5.1:compile (default-compile) on project closure-compiler: Compilation failure: Compilation failure: 
[ERROR] /home/centos/closure-compiler/src/com/google/javascript/refactoring/FixingErrorManager.java:[30,35] [deprecation] BasicErrorManager in com.google.javascript.jscomp has been deprecated
[ERROR] /home/centos/closure-compiler/src/com/google/javascript/jscomp/deps/JsFileFullParser.java:[27,35] [deprecation] BasicErrorManager in com.google.javascript.jscomp has been deprecated
[ERROR] /home/centos/closure-compiler/src/com/google/javascript/jscomp/ant/AntErrorManager.java:[19,35] [deprecation] BasicErrorManager in com.google.javascript.jscomp has been deprecated
[ERROR] /home/centos/closure-compiler/src/com/google/javascript/refactoring/FixingErrorManager.java:[30,35] [deprecation] BasicErrorManager in com.google.javascript.jscomp has been deprecated
```

My jdk version is different than yours:

`Java version: 1.8.0_242, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.242.b08-0.el7_7.x86_64/jre`

vs yours

`Java version: 1.8.0_242, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-8-openjdk/jre`

What OS are you running? I running on centos 7.

Also, could you share your ~/.m2/settings.xml?

Thanks.

**Comment by niloc132** [comment]
I'm on arch linux, but have built on a mac as well, will reply from there later to confirm that different everything still works over there. 

I'm not going to redact+post my settings.xml, instead I just moved it out of the way and rebuilt with no settings.xml at all, same result. I also switched JDK to the actual oracle one, see below for successful build log and jvm vers string (note though that it is older than yours):

```
[colin@runes closure-compiler (master)]$ mvn -DskipTests -pl externs/pom.xml,pom-main.xml,pom-main-shaded.xml  clean install
...
[INFO] --- maven-install-plugin:2.4:install (default-install) @ closure-compiler ---
[INFO] Installing /home/colin/workspace/closure-compiler/target/closure-compiler-1.0-SNAPSHOT.jar to /home/colin/.m2/repository/com/google/javascript/closure-compiler/1.0-SNAPSHOT/closure-compiler-1.0-SNAPSHOT.jar
[INFO] Installing /home/colin/workspace/closure-compiler/dependency-reduced-pom.xml to /home/colin/.m2/repository/com/google/javascript/closure-compiler/1.0-SNAPSHOT/closure-compiler-1.0-SNAPSHOT.pom
[INFO] Installing /home/colin/workspace/closure-compiler/target/closure-compiler-1.0-SNAPSHOT-sources.jar to /home/colin/.m2/repository/com/google/javascript/closure-compiler/1.0-SNAPSHOT/closure-compiler-1.0-SNAPSHOT-sources.jar
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Closure Compiler Externs 1.0-SNAPSHOT:
[INFO] 
[INFO] Closure Compiler Externs ........................... SUCCESS [  0.797 s]
[INFO] Closure Compiler Main .............................. SUCCESS [  0.465 s]
[INFO] Closure Compiler ................................... SUCCESS [ 27.226 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  29.059 s
[INFO] Finished at: 2020-02-06T10:22:25-06:00
[INFO] ------------------------------------------------------------------------
[colin@runes closure-compiler (master)]$ mvn -version
Apache Maven 3.6.3 (NON-CANONICAL_2019-11-27T20:26:29Z_root)
Maven home: /opt/maven
Java version: 1.8.0_181, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-8-jdk/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.4.15-arch1-1", arch: "amd64", family: "unix"
```
Perhaps you can gist/pastebin the full output of your clean install command? Your very first abrev'd log above included this `org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile`, but the other snippets showed that you were using maven-compiler-plugin 3.5.1 (which is what the poms in this project specify), so I'm wondering if something else is changed in your setup. 

I also confirmed that removing these two lines from the pom removed all the deprecation warnings (they don't seem to be configured as failures in the pom):
```
            <showDeprecation>true</showDeprecation>
            <compilerArgument>-Xlint:unchecked,deprecation,fallthrough,finally</compilerArgument>
```

**Comment by carylewis** [comment]
I was finally able to get the build to work. I was thinking that perhaps I was running into a memory limitation rather than a compile bug.

So, I modified the pom.xml, updated the compiler plugin to use 3.8.1 (probably not needed), changed fork to false.

Then I set 

`MAVEN_OPTS=-Xmx1024m`

and then the:

`mvn -X -DskipTests -pl externs/pom.xml,pom-main.xml,pom-main-shaded.xml  clean install`

worked.

Thanks for all of your help.
